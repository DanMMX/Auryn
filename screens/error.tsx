import * as React from 'react';
import { View, StyleSheet, Text } from 'react-native';

interface ErrorProps {
  text: string;
}

export const Error: React.FunctionComponent<ErrorProps> = ({ text }) => (
  <View style={styles.container}>
    <Text style={styles.text}>{text}</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderColor: 'black',
    borderWidth: 3,
    backgroundColor: '#143672',
    alignItems: 'center',
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
  text: {
    fontSize: 36,
    color: 'white',
  },
});
