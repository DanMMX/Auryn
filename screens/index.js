import Splash   from './splash.youi'
import Lander   from './lander.youi'
import PDP      from './pdp.youi'
import Player   from './video.youi'

export {
  Splash,
  Lander,
  PDP,
  Player,
}
