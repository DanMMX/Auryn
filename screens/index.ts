/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import Splash from './splash';
import Lander from './lander';
import PDP from './pdp';
import Search from './search';
import Profile from './profile';
import Video from './video';

export {
  Splash,
  Lander,
  PDP,
  Search,
  Profile,
  Video,
};
