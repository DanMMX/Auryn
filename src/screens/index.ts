/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

export { Splash }   from './splash';
export { Lander }   from './lander';
export { Pdp }      from './pdp';
export { Search }   from './search';
export { Profile }  from './profile';
export { VideoScreen as Video }    from './video';
