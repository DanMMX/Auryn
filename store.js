import { applyMiddleware, createStore } from 'redux';
import logger from 'redux-logger';
import promise from 'redux-promise-middleware';
import thunk from 'redux-thunk';

import reducer from './reducers';

const middleware = applyMiddleware(thunk, logger, promise());

export default createStore(reducer, middleware);
