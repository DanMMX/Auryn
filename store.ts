/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { applyMiddleware, createStore } from 'redux';
import promise from 'redux-promise-middleware';
import createDebounce from 'redux-debounced';
import thunk from 'redux-thunk';
import { rootReducer } from './reducers';

const middleware = applyMiddleware(createDebounce(), thunk, promise);

export default createStore(rootReducer, middleware);
