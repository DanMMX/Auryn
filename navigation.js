import * as Screens from './screens';
// CES
import { AdOverlay } from './ces2018';
// END CES
import { createStackNavigator, createSwitchNavigator } from 'react-navigation';

const stackOptions = {
  headerMode: 'none',
  cardStyle: {
    backgroundColor: 'transparent',
    opacity: 1,
  },
  transitionConfig: () => ({
    transitionSpec: {
      duration: 0,
    },
    containerStyle: {
      backgroundColor: 'transparent',
    },
    screenInterpolator: () => { },
  }),
};

const AppStack = createStackNavigator(
{
    Lander: { screen: Screens.Lander },
    PDP: { screen: Screens.PDP },
    Search: { screen: Screens.Search },
    Profile: { screen: Screens.Profile },
    Video: { screen: Screens.Video },
    // CES
    AdOverlay: { screen: AdOverlay },
    // END CES
  },
  {
    ...stackOptions,
    initialRouteName: 'Lander',
  }
);

const SplashStack = createStackNavigator(
  { Splash: { screen: Screens.Splash } },
  {
    ...stackOptions,
    initialRouteName: 'Splash',
  }
);

const rootNavigationStack = createSwitchNavigator({
  Splash: SplashStack,
  App: AppStack,
});

export default rootNavigationStack;
