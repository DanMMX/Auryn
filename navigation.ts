/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as Screens from './screens';
import { createStackNavigator, createSwitchNavigator, StackNavigatorConfig } from 'react-navigation';
import { Config } from './config';

const stackOptions: StackNavigatorConfig = {
  headerMode: 'none',
  cardStyle: {
    backgroundColor: 'transparent',
    opacity: 1,
  },
  transitionConfig: () => ({
    transitionSpec: {
      duration: 0,
    },
    containerStyle: {
      backgroundColor: 'transparent',
    },
    screenInterpolator: () => { },
  }),
};

const createNavigator = Config.isRoku ? createSwitchNavigator : createStackNavigator;
const AppStack = createNavigator(
  {
    Lander: { screen: Screens.Lander },
    PDP: { screen: Screens.PDP },
    Search: { screen: Screens.Search },
    Profile: { screen: Screens.Profile },
    Video: { screen: Screens.Video },
  },
  {
    ...stackOptions,
    initialRouteName: 'Lander',
  }
);

const SplashStack = createStackNavigator(
  { Splash: { screen: Screens.Splash } },
  {
    ...stackOptions,
    initialRouteName: 'Splash',
  }
);

const rootNavigationStack = createSwitchNavigator({
  Splash: SplashStack,
  App: AppStack,
});

export default rootNavigationStack;
