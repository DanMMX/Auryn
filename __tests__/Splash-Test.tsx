/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { Splash } from '../screens';
import { Provider } from 'react-redux';
import { navigationProp } from '../__mocks__/navigation';
import store from '../store';
import renderer from 'react-test-renderer';

test('renders correctly with error', () => {
  const tree = renderer
    .create(<Provider store={store}>
      <Splash {...navigationProp} error="Splash Screen with Error"/>
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}>
      <Splash {...navigationProp} />
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


