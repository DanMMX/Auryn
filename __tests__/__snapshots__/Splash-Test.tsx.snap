// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
>
  <Composition
    onTimelinesLoaded={[Function]}
    source="Auryn_Splash"
  >
    <TimelineRef
      autoplay={false}
      direction="forward"
      loop={false}
      name="SplashIn"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onCompositionDidLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
    />
    <TimelineRef
      autoplay={false}
      direction="forward"
      loop={false}
      name="SplashOut"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onCompositionDidLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
    />
    <ViewRef
      name="Loader"
      onCompositionDidLoad={[Function]}
      onTimelinesLoaded={[Function]}
    >
      <TimelineRef
        autoplay={true}
        direction="forward"
        loop={true}
        name="Loop"
        onAborted={[Function]}
        onCompleted={[Function]}
        onCompletedForward={[Function]}
        onCompletedReverse={[Function]}
        onCompositionDidLoad={[Function]}
        onPaused={[Function]}
        onPausedForward={[Function]}
        onPausedReverse={[Function]}
        onPlay={[Function]}
        onPlayForward={[Function]}
        onPlayReverse={[Function]}
        onStarted={[Function]}
        onStartedForward={[Function]}
        onStartedReverse={[Function]}
      />
    </ViewRef>
  </Composition>
</View>
`;

exports[`renders correctly with error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#143672",
      "borderColor": "black",
      "borderWidth": 3,
      "flex": 1,
      "flexWrap": "wrap",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 36,
      }
    }
  >
    Splash Screen with Error
  </Text>
</View>
`;
